src/sysext/sysext.c

```diff
@@ -510,7 +510,15 @@ static int merge_subprocess(Hashmap *images, const char *workspace) {
                                 }
                         }
 
-                        r = mount_nofollow_verbose(LOG_ERR, img->path, p, NULL, MS_BIND, NULL);
+                        /* Make individual confexts directories noexec, as code should be shipped via
+                         * sysexts. Note that we can't quite make the whole overlay noexec, as there are
+                         * still executable scripts shipped in the base /etc. */
+                        r = mount_nofollow_verbose(LOG_ERR,
+                                                   img->path,
+                                                   p,
+                                                   NULL,
+                                                   MS_BIND | (arg_image_class == IMAGE_CONFEXT ? MS_NOEXEC : 0),
```

Comment:
did you test this? iirc you have to apply the flags in a second mount() call.

Reply:
also, please maybe make this a table somewhere, indexed by the image class, and then set the other relevant flags too:

MS_NODEV on both
MS_NOSUID/MS_NOSYMFOLLOW on neither